### PR TITLE
Platform Event Queue Interface

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Defines a cross-platform event queue interface that will be later used to refactor the execution context logic.

## Testing

Unit test added

## Documentation

N/A

### CODE REVIEW

File: src/inc/quic_platform_winuser.h [hunk 1]
```diff
@@ -572,6 +572,67 @@ typedef HANDLE CXPLAT_EVENT;
 #define CxPlatEventWaitWithTimeout(Event, timeoutMs) \
     (WAIT_OBJECT_0 == WaitForSingleObject(Event, timeoutMs))
 
+//
+// Event Queue Interfaces
+//
+
+typedef HANDLE CXPLAT_EVENTQ;
+#define CXPLAT_SQE OVERLAPPED
```
Comment (maolson-msft) [inline src/inc/quic_platform_winuser.h:580]: What do "SQE" and "CQE" stand for?

Comment (nibanks) [reply src/inc/quic_platform_winuser.h:580]: "Submission queue event" and "completion queue event". Stolen from io_uring terminology.

File: src/platform/unittest/PlatformTest.cpp [hunk 2]
```diff
@@ -51,3 +51,73 @@ TEST(PlatformTest, QuicAddrParsing)
...
+    // Single queue/dequeue tests
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe1, &user_data1));
+    ASSERT_EQ(1u, CxPlatEventQDequeue(&queue, events, 2, 0));
+    ASSERT_EQ((void*)&user_data1, CxPlatCqeUserData(&events[0]));
+
+    // Multiple queue/dequeue tests
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe1, &user_data1));
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe2, &user_data2));
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe3, &user_data3));
+    ASSERT_EQ(2u, CxPlatEventQDequeue(&queue, events, 2, 100));
```
Comment (mtfriesen) [inline src/platform/unittest/PlatformTest.cpp:84]: why wait 100ms for the first 2 completions, but 0 ms for the third completion?

**Review by mtfriesen — APPROVED** [review]

File: src/platform/unittest/PlatformTest.cpp (hunk 2 above)
Comment (nibanks) [reply src/platform/unittest/PlatformTest.cpp:84]: Just for the sake of changing things up. No specific reason.

**Review by liveans — APPROVED** [review]

**Pull request merged**